sympy/core/power.py

```diff
@@ -579,9 +580,12 @@ def _check(ct1, ct2, old):
                     pow = as_int(pow)
                     combines = True
                 except ValueError:
-                    combines = Pow._eval_power(
-                        Pow(*old.as_base_exp(), evaluate=False),
-                        pow) is not None
+                    if denom(pow) == 1:
```

Comment:
What if 

```
In [36]: g = 6**(S(2)/3 *x) + I

In [37]: g.subs(6**(x), y)
Out[37]: 
 2⋅x    
 ───    
  3     
6    + ⅈ
```
problem is at line 580

```
In [12]: as_int(2*pi)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-12-bc65b46ebd66> in <module>()
----> 1 as_int(2*pi)

/root/sympy/sympy/core/compatibility.pyc in as_int(n)
    334             raise TypeError
    335     except TypeError:
--> 336         raise ValueError('%s is not an integer' % n)
    337     return result
    338 

ValueError: 2*pi is not an integer

In [13]: as_int(2)
Out[13]: 2
```

I don't know why there is no testcases for rational, Imaginary power.